src/pages/billing/SpendReport/SpendReport.test.ts

```diff
@@ -126,54 +132,58 @@ describe('SpendReport', () => {
     } as Partial<AjaxContract> as AjaxContract))
-    getSpendReport.mockResolvedValueOnce(
-      createSpendReportResult('1110')
-    ).mockResolvedValue(
-      createSpendReportResult('1110.17')
-    )
+    getSpendReport.mockResolvedValue(createSpendReportResult('1110'))
 
     // Act
     await act(async () => {
       await render(h(SpendReport, { viewSelected: true, billingProjectName: 'thrifty' }))
     })
-    await select90Days()
 
     // Assert
-    expect(screen.getByTestId('spend').textContent).toBe('$1,110.17*')
-    screen.getByText(otherCostMessaging)
-    expect(getSpendReport).toHaveBeenCalledTimes(2)
-    expect(getSpendReport).toHaveBeenNthCalledWith(1, { billingProjectName: 'thrifty', endDate: '2022-04-01', startDate: '2022-03-02' })
-    expect(getSpendReport).toHaveBeenNthCalledWith(2, { billingProjectName: 'thrifty', endDate: '2022-04-01', startDate: '2022-01-01' })
+    await waitFor(() => {
+      expect(screen.getByText(/\$89.00 in other infrastructure/i)).toBeInTheDocument()
+    })
+    expect(getSpendReport).toHaveBeenCalledTimes(1)
+    expect(screen.getByTestId('spend').textContent).toBe('$1,110.00*')
+    expect(screen.getByTestId('compute').textContent).toBe('$999.00')
+    expect(screen.getByTestId('storage').textContent).toBe('$22.00')
+
+    // Highcharts content is very minimal when rendered in the unit test. Testing of "most expensive workspaces"
+    // is in the integration test. Accessibility is also tested in the integration test.
   })
 
-  it('displays cost information', async () => {
+  it('fetches reports based on selected date range, if active', async () => {
     //Arrange
     const getSpendReport = jest.fn()
     asMockedFn(Ajax).mockImplementation(() => ({
       Billing: { getSpendReport } as Partial<AjaxContract['Billing']>
     } as Partial<AjaxContract> as AjaxContract))
-    getSpendReport.mockResolvedValue(createSpendReportResult('1110'))
+    getSpendReport.mockResolvedValueOnce(
+      createSpendReportResult('1110')
+    ).mockResolvedValue(
+      createSpendReportResult('1110.17')
+    )
 
     // Act
     await act(async () => {
       await render(h(SpendReport, { viewSelected: true, billingProjectName: 'thrifty' }))
     })
+    await select90Days()
 
     // Assert
-    expect(getSpendReport).toHaveBeenCalledTimes(1)
-    expect(screen.getByTestId('spend').textContent).toBe('$1,110.00*')
-    expect(screen.getByTestId('compute').textContent).toBe('$999.00')
-    expect(screen.getByTestId('storage').textContent).toBe('$22.00')
-    screen.getByText(/\$89.00 in other infrastructure/i)
-
-    // Highcharts content is very minimal when rendered in the unit test. Testing of "most expensive workspaces"
-    // is in the integration test. Accessibility is also tested in the integration test.
+    await waitFor(() => {
+      expect(screen.getByText(otherCostMessaging)).toBeInTheDocument()
+    })
+    expect(screen.getByTestId('spend').textContent).toBe('$1,110.17*')
```

Comment:
Nit: could use [toHaveTextContent](https://github.com/testing-library/jest-dom#tohavetextcontent). Same comment for other tests.

```suggestion
    expect(screen.getByTestId('spend')).toHaveTextContent('$1,110.17*')
```